bmwqemu.pm

```diff
@@ -227,6 +226,7 @@ sub modstate {
 
 sub current_test {
     require autotest;
+    no warnings 'once';
```

Comment:
I can't remember right now if there is a better fix. a warning doesn't appear when using `use` instead of require.

Reply:
Wonder if the `require` above should be moved to the top of the file, or if it really should be loaded very late at runtime.

Reply:
Actually `autotest` does `use bmwqemu`, so this is another example of a circular dependency, and doing the `require autotest` at runtime at least avoids compile time warnings.

I don't see a quick fix for this, as we are exposing the `$autotest::current_test` variable as a global var *sigh*.
(Looking into os-autoinst-distri-opensuse, I can see it used only in 3 files, but still, we don't know what other code is out there using it. I guess there should have been a testapi function providing it.)
The correct fix for this circular dependency would be to move the current_test name into its own module, that `autotest` and `bmwqemu` can use.
To not break code, this could be solved with making `$autotest::current_test` a tied scalar, setting the real value in the `STORE` function call.

Unless we want to invest time with this right now, I would say we keep the `no warnings` here for now.

Reply:
I created https://progress.opensuse.org/issues/104670 for this one